bangazonapi/views/profile.py

```diff
@@ -275,13 +275,38 @@ def favoritesellers(self, request):
             ]
         """
         customer = Customer.objects.get(user=request.auth.user)
-        favorites = Favorite.objects.filter(customer=customer)
+        if request.method == "GET":
+
+           
+            favorites = Favorite.objects.filter(customer=customer)
 
-        serializer = FavoriteSerializer(
-            favorites, many=True, context={'request': request})
-        return Response(serializer.data)
+            serializer = FavoriteSerializer(
+                favorites, many=True, context={'request': request})
+            return Response(serializer.data)
 
 
+        if request.method == "POST":
+            seller_id = self.request.data["seller"]
+            if seller_id is not None:
+                seller = Customer.objects.get(pk=seller_id)
+                try:
+                    favorite = Favorite.objects.get(customer=customer, seller=seller)
+                except Favorite.DoesNotExist as ex:
+                    new_favorite = Favorite()
+                    new_favorite.customer = customer
+                    new_favorite.seller = seller
+                    new_favorite.save()
+        
+                    serializer = FavoriteSerializer(
+                        new_favorite, context={'request': request})
+                    return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+                serializer = FavoriteSerializer(
+                        favorite, context={'request': request})
+                return Response(serializer.data, status=status.HTTP_201_CREATED)
```

Comment:
So if an existing `Favorite` is found on line 293, you're still returning a `201_CREATED`?